guided-match/ddl.sql

```diff
@@ -0,0 +1,204 @@
+            in a separate script
+
+
+*/
+
+/* Neo4J
+CREATE TABLE ccs_outcomes (
+   outcome_id                       INTEGER PRIMARY KEY,
+   journey_question_id              INTEGER NOT NULL,
+   outcome_evaluator_code           VARCHAR(10) NOT NULL, 
+   journey_answer_id                INTEGER NOT NULL,
+   ccs_support_ind                  CHAR(1) NOT NULL
+);
+*/
+
+/* This table creation script belongs in the agreement ddl.sql
+CREATE TABLE commercial_agreements (
+  commercial_agreement_id           INTEGER PRIMARY KEY,
+  commercial_agreement_number       VARCHAR(20) NOT NULL,
+  commercial_agreement_name         VARCHAR(200) NOT NULL,
+  commercial_agreement_owner        VARCHAR(200) NOT NULL,
+  description                       VARCHAR(2000) NOT NULL, -- perhaps this needs to be a different type
+  start_date                        DATE NOT NULL,
+  end_date                          DATE NOT NULL,
+  agreement_url                     VARCHAR(2000) NOT NULL,
+  authorisation_email               VARCHAR(200)
+);
+CREATE INDEX COAG_IDX1 on COMMERCIAL_AGREEMENTS  (commercial_agreement_number);
+CREATE INDEX COAG_IDX2 on COMMERCIAL_AGREEMENTSS (commercial_agreement_name);
+CREATE INDEX COAG_IDX3 on COMMERCIAL_AGREEMENTS  (commercial_agreement_owner);
+CREATE INDEX COAG_IDX4 on COMMERCIAL_AGREEMENTS  (authorisation_email);
+*/
+
+/* Neo4J 
+CREATE TABLE domain_question_outcomes (
+   outcome_id                       INTEGER PRIMARY KEY,
+   journey_question_id              INTEGER NOT NULL,
+   outcome_evaluator_code           VARCHAR(10) NOT NULL, 
+   journey_answer_id                INTEGER NOT NULL,
+   modifier_journey_name            VARCHAR(10) NOT NULL
+);
+*/
+/* Neo4J
+CREATE TABLE explicit_question_outcomes (
+   outcome_id                       INTEGER PRIMARY KEY,
+   journey_question_id              INTEGER NOT NULL,
+   outcome_evaluator_code           VARCHAR(10) NOT NULL, 
+   journey_answer_id                INTEGER NOT NULL,
+   resulting_journey_question_id    INTEGER NOT NULL
+);
+*/
+
+/* Neo4J
+CREATE TABLE journey_answers ( --Need to check this one out with Dave H as attributes on model are different to descriptions on confluence page
+  journey_answer_id                INTEGER PRIMARY KEY,
+  journey_question_id              INTEGER NOT NULL,
+  answer_type                      VARCHAR(50),
+  answer_label                     VARCHAR(100),
+  default_ind                      CHAR(1), -- Y or N default value will be N as only one can be Y
+  valid_value                      VARCHAR(200),
+  url                              VARCHAR(200), -- notsure required anymore
+  regex_validation                 VARCHAR(2000)  -- notsure required anymore
+);
+*/
+
+CREATE TABLE journey_instance_questions (
+  journey_instance_question_id      INTEGER PRIMARY KEY,
+  journey_question_id               INTEGER NOT NULL,

```

Comment:
Question - is `journey_question_id` here meant to be `journey_instance_id` (and a foreign key back to `journey_instances`)?  Probably also need:

- `UUID` type column to accommodate a reference to the question
- `hint` (text)
- This model does not currently appear to accommodate multi-select questions (but the service, graph and [API](https://raw.githubusercontent.com/Crown-Commercial-Service/ccs-scale-api-definitions/master/guided-match/guided-match-service.yaml) do - see `QuestionHistory` and `AnswerHistory` schema types)
- `answer_id` needs to be type `UUID` to reference the corresponding answer node in the graph

Reply:
Yes needs a fk back to journey instance. Will add.

Reply:
Not sure what hint relates to. Will check with you. I have added fk to journey_iinstances and made it type UUID. I have chnaged type of answer_id to UUID and changed the type of journey_question_id to UUID.

Regarding multi select questions I will have to think about this. I think it needs an extra table.

Reply:
Talking to Dave Hughes we don't think that this table needs a hint column as this table is a history table. I do agree about multiple answers and need to look into this.

Reply:
> Talking to Dave Hughes we don't think that this table needs a hint column as this table is a history table. I do agree about multiple answers and need to look into this.

Agreed - must have been coffee deprived when I put that comment in!